### PR TITLE
Using IHubContext in hub agnostic code

### PR DESCRIPTION
**Pull request by BrennanConroy**
Another preview7 feature

[Internal Review Link](https://review.docs.microsoft.com/en-us/aspnet/core/signalr/hubcontext?branch=pr-en-us-22943&view=aspnetcore-6.0#use-ihubcontext-in-generic-code)

### CODE REVIEW

File: aspnetcore/signalr/hubcontext.md [hunk 1]
```diff
@@ -11,10 +11,11 @@ uid: signalr/hubcontext
 ---
 # Send messages from outside a hub
 
-By [Mikael Mengistu](https://twitter.com/MikaelM_12)
-
 The SignalR hub is the core abstraction for sending messages to clients connected to the SignalR server. It's also possible to send messages from other places in your app using the `IHubContext` service. This article explains how to access a SignalR `IHubContext` to send notifications to clients from outside a hub.
 
+> [!NOTE]
+> The `IHubContext` is for sending notifications to clients, it is not used to call methods on the `Hub`.
```
Comment (BrennanConroy) [inline aspnetcore/signalr/hubcontext.md:17]: So many people ask on Stackoverflow how to call hub methods from controllers...

**Review by bradygaster — APPROVED** [review]

File: aspnetcore/signalr/hubcontext.md [hunk 2]
```diff
@@ -101,6 +102,40 @@ public class ChatController : Controller
...
+    MethodB((IHubContext<Hub>)hubContext);
+
+    MethodC((IHubContext)hubContext);
+
+    await next.Invoke();
+}
+```
+
+This is useful when:
+* Writing libraries that don't have a reference to the specific `Hub` type your application is using
```
Comment (Rick-Anderson) [inline aspnetcore/signalr/hubcontext.md:134]:
```suggestion
* Writing libraries that don't have a reference to the specific `Hub` type the app is using.
```

**Review by Rick-Anderson — APPROVED** [review]

File: aspnetcore/signalr/hubcontext.md (hunk 2 above)
Comment (BrennanConroy) [reply aspnetcore/signalr/hubcontext.md:134]: Is "app" or "application" preferred?

Comment (Rick-Anderson) [reply aspnetcore/signalr/hubcontext.md:134]: The style guide prefer app.

File: aspnetcore/signalr/hubcontext.md [hunk 3]
```diff
@@ -101,6 +102,40 @@ public class ChatController : Controller
...
+
+app.Use(async (context, next) =>
+{
+    var hubContext = context.RequestServices
+                            .GetRequiredService<IHubContext<ChatHub>>();
+    MethodA((IHubContext<BaseHub>)hubContext);
+
+    MethodB((IHubContext<Hub>)hubContext);
+
+    MethodC((IHubContext)hubContext);
```
Comment (halter73) [inline aspnetcore/signalr/hubcontext.md:127]: Why even bother demonstrating that the `(IHubContext<BaseHub>)` and `(IHubContext<Hub>)` casts are possible? It just adds unnecessary length to the conceptual docs when you can do all the same things by casting to `(IHubContext)` instead.

If someone really cares that it's covariant, they'll be able to see the `out THub` in the API docs, but it's suuuper niche.

**Review by halter73 — APPROVED** [review]

**Pull request merged**